x-pack/test/functional/apps/security/users.ts

```diff
@@ -115,5 +118,104 @@ export default function ({ getService, getPageObjects }: FtrProviderContext) {
       expect(roles.monitoring_user.reserved).to.be(true);
       expect(roles.monitoring_user.deprecated).to.be(false);
     });
+
+    describe('edit', function () {
+      before(async () => {
+        await PageObjects.security.clickElasticsearchUsers();
+      });
+
+      describe('update user profile', () => {
+        it('when submitting form and redirects back', async () => {
+          const user = {
+            username: 'OptionalUser',
+            full_name: 'Optional User',
+            email: 'optionalUser@elastic.co',
+            roles: [],
+          };
+          await PageObjects.security.updateUser(user);
+          const users = keyBy(await PageObjects.security.getElasticsearchUsers(), 'username');
+
+          expect(users.OptionalUser.roles).to.eql(['']);
+          expect(users.OptionalUser.fullname).to.eql(user.full_name);
+          expect(users.OptionalUser.email).to.eql(user.email);
+          expect(users.OptionalUser.reserved).to.be(false);
+          expect(await browser.getCurrentUrl()).to.contain('app/management/security/users');
+        });
+      });
+
+      describe('change password', () => {
+        before(async () => {
+          await toasts.dismissAllToasts();
+        });
+        afterEach(async () => {
+          await PageObjects.security.submitUpdateUserForm();
+          await toasts.dismissAllToasts();
+        });
+        it('of other user when submitting form', async () => {
+          const user = {
+            username: 'OptionalUser',
+            new_password: 'NewOptionalUserPwd',
+            confirm_password: 'NewOptionalUserPwd',
+            roles: [],
+          };
+          await PageObjects.security.updateUserPassword(user);
+          await retry.waitFor('', async () => {
+            const toastCount = await toasts.getToastCount();
+            return toastCount >= 1;
+          });
+          const succesToast = await toasts.getToastElement(1);
+          expect(await succesToast.getVisibleText()).to.be("Password changed for 'OptionalUser'.");
+        });
+
+        it('of current user when submitting form', async () => {
+          const user = {
+            username: 'test_user',
+            password: 'changeme',
```

Comment:
I did some digging to see exactly where this user's credentials are defined. It's here: 

https://github.com/elastic/kibana/blob/60340b5a05d5936343f6f7cd97628623cfd3834a/test/common/services/security/test_user.ts#L18-L19

Docs: https://www.elastic.co/guide/en/kibana/8.0/development-tests.html#_using_the_test_user_service

However, we have a different `adminTestUser` that is used in our security API integration tests:

https://github.com/elastic/kibana/blob/60340b5a05d5936343f6f7cd97628623cfd3834a/x-pack/test/security_api_integration/tests/session_invalidate/invalidate.ts#L10

These are defined by the `@kbn/test` package, they can apparently be changed with environment variables and may be different for Cloud testing: 
https://github.com/elastic/kibana/blob/6693ef371f887eca639b09c4c9b15701b4ebabd4/packages/kbn-test/src/kbn/users.ts#L9-L24

----

All that said, using the hardcoded `test_user` / `changeme` in this file makes me a bit nervous because 1. I'm not sure if the tests will always be executed with the user defined in `test_user.ts`, and 2. even so, if that user changes, then this test will fail.

So, maybe we should just export the test username/password from that file so it can be used here, and just call it a day. But I want to know what @azasypkin thinks, in case I'm missing anything else RE: Cloud testing.

Reply:
> All that said, using the hardcoded test_user / changeme in this file makes me a bit nervous

Yeah, I share the same concern, but mostly because I treat service test users as immutable to avoid any side effects for the following tests. Is there any reason why we cannot do something like this (semi pseudo-code)?
```ts
const user = {
  username: 'throw-away-user',
  password: 'changeme',
  new_password: 'changeme',
  confirm_password: 'changeme',
  roles: ['kibana_admin'],
};

// The API integration tests service to create through API, not UI
await security.user.create(user.username, {
  password: user.password,
  roles: user.roles,
  full_name: 'Admin',
});

// Re-login (should be pretty stable)
await PageObjects.security.forceLogout();
await PageObjects.security.loginSelector.login('basic', 'basic', { 
  username: user.username, password: user.password
});

await PageObjects.security.updateUserPassword(user, true);

// Make sure new password works.
await PageObjects.security.forceLogout();
await PageObjects.security.loginSelector.login('basic', 'basic', { 
  username: user.username, password: user.new_password
});
```

Reply:
I was just trying to be lazy here :) since I was not changing the password, but if everyone is concerned about it, I will just login using optionalUser. Not a problem

Reply:
Yeah my main concern is that we might break Cloud tests. If we do that, we don't find out until after the fact, and it's just a whole thing. Safer to do what Oleg suggested.

Reply:
Also I am not going to test if the new password works, I am assuming that we should have API integration for that. What I want to test here that when I am logging as a current user and I am changing my password, I will a text field asking for my current password and then success.

Reply:
Agreed, that makes sense to me

Reply:
> What I want to test here that when I am logging as a current user and I am changing my password, and I will have a text field asking for my current password and then I can change my password. We do not want to do too much in one test.

Sounds good to me as well. Would you mind keeping this comment in the test as well? It'd help future readers to understand what we want to test exactly here.